### PR TITLE
[WebProfilerBundle] Improve cache panel

### PR DESCRIPTION
**Pull request by ro0NL**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #... 
| License       | MIT
| Doc PR        | symfony/symfony-docs#... 

An attempt to improve the page a bit. Personally i think all elements on a single page is too much info.

Before

image

image

After

image

image

### CODE REVIEW

**Review by javiereguiluz — COMMENTED** [review]
👍  trying to improve this page was on my TODO list for 3.3 too. @ro0NL thanks for taking care of this.

I agree that this page has too much information. Looking at your screenshot, I'd like to suggest these additional changes:

1) Use the `<span class="unit">ms</span>` code to display the "ms" value, as we do in other profiler panels such as "Performance" (and maybe apply it to the "%" too in "hit/misses" ??)

2) Rearrange the row of information as follows:

* Total calls
* Total time
* `<div class="metric-divider"></div>`
* Total reads
* Total writes
* Total deletes
* `<div class="metric-divider"></div>`
* Total Hits
* Total Misses
* Hits/reads

**Comment by ro0NL** [comment]
Improved the metric handling, moving things to the view layer. Totally worth it :) (screenshot updated).

File: src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/cache.html.twig [hunk 1]
```diff
@@ -54,93 +54,115 @@
...
-                        <th style="padding-top:2rem">Pool::{{ call.name }}</th>
-                    </tr>
-                    <tr>
-                        <th>Argument</th>
-                        <td>{{ profiler_dump(call.argument, maxDepth=2) }}</td>
-                    </tr>
-                    <tr>
-                        <th>Results</th>
-                        <td>
-                            {% if call.result != false %}
```
Comment (ro0NL) [inline src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/cache.html.twig]: @Nyholm not sure about a) this check and b) loose comparison. As you can see in the screenshots `null` was hidden as wel. Does `false` need to be excluded, strictly?

Comment (Nyholm) [inline src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/cache.html.twig]: Thank you for this PR. 
I'm not sure about your questions. Call.result could be false if there was an error (if I remember correctly). I **think** that is the reason for this check. 

Why it not is a strict comparison... that is probably a typo.

**Review by stof — APPROVED** [review]
👍

File: src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/cache.html.twig (hunk 1 above)
Comment (ro0NL) [inline src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/cache.html.twig]: I'll check it out tonight. I'd prefer something like displaying `n/a` in that case, opposed to _just_ an empty cell.

Comment (ro0NL) [inline src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/cache.html.twig]: @Nyholm figured it out; this is (AFAIK) the direct result value from the method call passed by the traceableadapter.

Now given https://github.com/php-fig/cache/blob/master/src/CacheItemInterface.php#L42 i dont think we should maky _any_ assumptions.

Besides, this was hides `false` as a result from a `hasItem` call ;-)

**Comment by ro0NL** [comment]
Did some overall improvements to the table structure.. i kinda like it. The panel looks really calm now and we save a lot on spacing overall.

Screenshots updated.

Should be reviewed with [`?w=1`](https://github.com/symfony/symfony/pull/22129/files?w=1)

**Comment by fabpot** [comment]
Thank you @ro0NL.

**Pull request closed without merging**